Missing ThisThread:: in get_name() of ThisThread.cpp

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect
undefined reference to `rtos::ThisThread::get_name()'
Linking error due to missing of **ThisThread::** in front of get_name()

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->


#### Target(s) affected by this defect ?
All

#### Toolchain(s) (name and version) displaying this defect ?
GCC_ARM. arm-none-eabi version 9.2.1

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->
b6370b4c37f3d4665ed1cdcb1afea85396bba1b3

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)
mbed-cli version 1.10.2

#### How is this defect reproduced ? 
call ThisThread::get_name() method

--- a/TESTS/mbedmicro-rtos-mbed/threads/main.cpp
+++ b/TESTS/mbedmicro-rtos-mbed/threads/main.cpp
@@ -500,9 +500,8 @@ void test_thread_name()
 
     const char tname[] = "Amazing thread";
     Thread t(osPriorityNormal, THREAD_STACK_SIZE, NULL, tname);
-    t.start(callback(thread_wait_flags));
     TEST_ASSERT_EQUAL(strcmp(tname, t.get_name()), 0);
-    t.flags_set(0x1);
+    t.start([&] { TEST_ASSERT_EQUAL(strcmp(tname, ThisThread::get_name()), 0); });
     t.join();
 }
 

--- a/rtos/source/ThisThread.cpp
+++ b/rtos/source/ThisThread.cpp
@@ -272,7 +272,7 @@ osThreadId_t ThisThread::get_id()
 #endif
 }
 
-const char *get_name()
+const char *ThisThread::get_name()
 {
 #if MBED_CONF_RTOS_PRESENT
     osThreadId_t id = osThreadGetId();

